### PR TITLE
[#159022591] Add Asset specs when adding an asset

### PR DESCRIPTION
**Pull request by zmwas**
## What does this PR do?
Adds specs for processor type, processor speed, screen size, storage and memory.

## Description of Task to be completed?
- Fix load asset category reducer
- Create asset spec radio buttons

## How should this be manually tested?
- Click on the add asset button
- Filter through the asset filter dropdowns and click next
- You should be able to save an asset with asset specs.

## What are the relevant pivotal tracker stories?
[159022591](https://www.pivotaltracker.com/story/show/159022591)

## Any background context you want to add?
N/A
## Important notes
N/A

## Packages installed
N/A

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
N/A

## Screenshots
screen shot 2018-08-01 at 11 34 41

### CODE REVIEW

**Comment by HawiCaesar** [comment]
@egichuri This is how it looks like when we add the changes you have requested.

screen shot 2018-07-31 at 16 36 28

File: src/components/Assets/AddAssetComponent.jsx [hunk 1]
```diff
@@ -6,6 +6,15 @@ import DropdownComponent from '../common/DropdownComponent';
 import InputFluid from '../common/TextInputComponent';
 import ArtButton from '../common/ButtonComponent';
 
+const numberOfYears = () => (
+  (new Date()).getFullYear() - 2012
+);
+
+const yearOfManufacture = Array(numberOfYears()).fill({}).map((value, index) => ({
```
Comment (sachungo) [inline src/components/Assets/AddAssetComponent.jsx]: The fact that `value` is not being used is an indication that there's a better way to do this. What are you trying to achieve with `numberOfYears` and `yearOfManufacture` functions in lay man's words? Have a look at [momentjs](https://momentjs.com/docs/) to see if it will simplify things

File: src/_components/Assets/AddAssetContainer.jsx [hunk 2]
```diff
@@ -156,17 +186,31 @@ class AddAssetContainer extends React.Component {
...
+          memory={this.state.memory}
+          onSelectYearOfManufacture={this.onSelectYearOfManufacture}
+          onSelectProcessorType={this.onSelectProcessorType}
+          onSelectProcessorSpeed={this.onSelectProcessorSpeed}
+          onSelectScreenSize={this.onSelectScreenSize}
+          onSelectStorage={this.onSelectStorage}
+          onSelectMemory={this.onSelectMemory}
         /> : <FilterAssetComponent
           {...this.props}
+          handleDropdownChanges={this.handleDropdownChanges}
```
Comment (sachungo) [inline src/_components/Assets/AddAssetContainer.jsx]: The render function of this component is becoming unwieldy due to the ternary operator and numerous props. Have you considered if things can be broken down to smaller components? For starters, use an if instead of the ternary operator. I.e.
```
render() {
  if (this.state.page === 1) {
    return (
      <AddAssetComponent
      ...
      ...
      />
    );
  }

  return (
    <FilterAssetComponent
      ...
      ...
    />
  );
}
```

File: src/_css/LoginComponent.css [hunk 3]
```diff
@@ -152,6 +152,10 @@ input,
     border: 0.5px solid #0a26a8;
     border-radius: 0; }
 
+.ui.form .inline.fields {
+  width: 340px;
+  padding: 3% 0; }
```
[the remark this answers is not in the document]
Comment (HawiCaesar) [reply src/_css/LoginComponent.css]: This is auto generated by the sass compiler to css. I think we accidentally committed this file.

File: .eslintrc [hunk 4]
```diff
@@ -105,6 +105,7 @@
     "jsx-a11y/label-has-for": 0,
     "linebreak-style": 0,
     "react/require-default-props": 0,
-    "react/sort-comp": 0
+    "react/sort-comp": 0,
+    "no-unused-vars": ["error", { "vars": "all", "args": "after-used", "ignoreRestSiblings": false, argsIgnorePattern: "event" }]
```
[the remark this answers is not in the document]
Comment (HawiCaesar) [reply .eslintrc]: There are some function [here]() that use the variable `event` so that we can pick user data. When we try and commit, we get `unused variable event` error. So we ignore the error specifically for the `event` variable.

File: src/_components/Assets/AddAssetContainer.jsx (hunk 2 above)
Comment (HawiCaesar) [reply src/_components/Assets/AddAssetContainer.jsx]: @zmwas and I had the same concern you are raising , we are looking into a more efficient way of doing this.

File: src/components/Assets/AddAssetComponent.jsx (hunk 1 above)
Comment (zmwas) [reply src/components/Assets/AddAssetComponent.jsx]: We wanted an array with an object for the years 2013 - the current year. We saw it would be quicker to do it this way and not use a library. I'll delete value and just use index.

Comment (zmwas) [reply src/components/Assets/AddAssetComponent.jsx]: @sachungo without value the dropdown won't have a key and we get the error 
`Encountered two children with the same key ReactJS`

Comment (zmwas) [reply src/components/Assets/AddAssetComponent.jsx]: screen shot 2018-08-01 at 3 58 30 pm

Comment (sachungo) [reply src/components/Assets/AddAssetComponent.jsx]: Ok...and please ensure it still works after the change

File: src/_components/Assets/AddAssetContainer.jsx (hunk 2 above)
Comment (zmwas) [reply src/_components/Assets/AddAssetContainer.jsx]: @sachungo we have refactored the add asset component. It now has SpecComponent as a child and the props are passed as an object.

File: src/_css/AddAssetComponent.css [hunk 5]
```diff
@@ -1,12 +1,8 @@
-.form-wrapper {
-  width: 100%;
-}
+.optional-label-text {
+  margin: 3% 0; }
```
[the remark this answers is not in the document]
Comment (HawiCaesar) [reply src/_css/AddAssetComponent.css]: we are using `scss` which is transpiled down to `css`

**Comment by HawiCaesar** [comment]
As per the radio section, the processor speed radio section was inline with the processor type radio section.
screen shot 2018-08-02 at 17 33 32

We are working on making the input labels closer to the input it relates to.

Making the width 100% over flows the radio section on to the darker bit of the modal. Like below
screen shot 2018-08-02 at 18 00 54

File: src/components/Assets/AddAssetComponent.jsx [hunk 6]
```diff
@@ -23,36 +24,31 @@ const AddAssetComponent = props => (
       <div className="page-text">Fill out device specs</div>
     </div>
     <Form onSubmit={props.onCreateAsset}>
-      <label className="label-style">Model</label>
+      <div className="label-style">Model Number</div>
```
Comment (sachungo) [inline src/components/Assets/AddAssetComponent.jsx]: Out of curiosity, why the change from label to div?

Comment (HawiCaesar) [reply src/components/Assets/AddAssetComponent.jsx]: At first we tried a `span` tag but the class `label-style` was not being applied. It very curiosu why it did not work. We finally decided on a `div`

**Comment by HawiCaesar** [comment]
About the sass setup, @Arthur236 and I did do a sass setup. I think I did a `git add src/` and forgot to remove the css files.
